TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/CapturedTrafficToHttpTransactionAccumulator.java

```diff
@@ -73,16 +85,37 @@ public void accept(String nodeId, String connectionId, TrafficObservation observ
                 throw new RuntimeException("Apparent out of order exception - " +
                         "found a purported write to a socket before a read!");
             }
+            log.trace("Adding response data for accum[" + connectionId + "]=" + accum);
             runningList.addResponseData(timestamp, observation.getWrite().getData().toByteArray());
+            log.trace("Added response data for accum[" + connectionId + "]=" + accum);
         } else if (observation.hasReadSegment()) {
             var accum = getAccumulationForFirstRequestObservation(nodeId, connectionId, timestamp);
             assert accum.state == Accumulation.State.NOTHING_SENT;
-            throw new RuntimeException("Not implemented yet.");
+            log.trace("Adding request segment for accum[" + connectionId + "]=" + accum);
+            if (accum.rrPair.requestData == null) {
+                accum.rrPair.requestData = new HttpMessageAndTimestamp(timestamp);
+            }
+            accum.rrPair.requestData.addSegment(observation.getReadSegment().getData().toByteArray());
+            log.trace("Added request segment for accum[" + connectionId + "]=" + accum);
         } else if (observation.hasWriteSegment()) {
             var accum = liveStreams.get(nodeId, connectionId, timestamp);
             assert accum != null && accum.state == Accumulation.State.REQUEST_SENT;
-            var runningList = accum.rrPair;
-            throw new RuntimeException("Not implemented yet.");
+            log.trace("Adding response segment for accum[" + connectionId + "]=" + accum);
+            if (accum.rrPair.responseData == null) {
+                accum.rrPair.responseData = new HttpMessageAndTimestamp(timestamp);
+            }
+            accum.rrPair.responseData.addSegment(observation.getWrite().getData().toByteArray());
+            log.trace("Added response segment for accum[" + connectionId + "]=" + accum);
+        } else if (observation.hasSegmentEnd()) {
+            var accum = liveStreams.get(nodeId, connectionId, timestamp);
+            assert accum != null && accum.state == Accumulation.State.REQUEST_SENT;
+            if (accum.rrPair.requestData.hasInProgressSegment()) {
```

Comment:
If there is ever a case where we process a response segment end before a request segment end this could be problematic, this shouldn't happen but makes me a little cautious. I'm almost wondering if we should create another task to split our EOM (or segment EOM) into either a read EOM or write EOM...

Reply:
There aren't any more write EOMs, so we could just change EOM to be EndOfRequest.

Reply:
Sorry -your right. We should do separate segment ends for both, or get rid of them in favor of an optional field up front to show a count

Reply:
[Here's a placeholder Jira for this](https://opensearch.atlassian.net/browse/MIGRATIONS-1174)

Reply:
Also though - it won't be EOM for a segment - more like, "end of a sequence of read segments" and "end of a sequence of write segments".  Those end of segments indicate the completion of a packet.  EOM indicates that the HttpMessage (now request) is now complete.